src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -0,0 +1,233 @@
+package org.broadinstitute.hellbender.tools.walkers.genotyper;
+
+import htsjdk.variant.variantcontext.*;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.collections.Permutation;
+import org.broadinstitute.hellbender.utils.genotyper.IndexedAlleleList;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+import org.broadinstitute.hellbender.utils.variant.GATKVariantContextUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Utilities class containing methods for restricting {@link VariantContext} and {@link GenotypesContext} objects to a
+ * reduced set of alleles, as well as for choosing the best set of alleles to keep and for cleaning up annotations and
+ * genotypes after subsetting.
+ *
+ * @author David Benjamin &lt;davidben@broadinstitute.org&gt;
+ */
+public final class AlleleSubsettingUtils {
+    private AlleleSubsettingUtils() {}  // prevent instantiation
+    private static final int PL_INDEX_OF_HOM_REF = 0;
+    private static final int MAX_LENGTH_FOR_PL_LOGGING = 100; // if PL vectors longer than this # of elements, don't log them
+
+    private static final GenotypeLikelihoodCalculators GL_CALCS = new GenotypeLikelihoodCalculators();
```

Comment:
Mmmm... perhaps I meant for GenptypeLikelihoodCalculators to be non-instantiable? Please fix that and use ```GenotypeLikelihoodCalculators.``` when accessing its static methods like ```getInstance```.